packages/chain-bridge/src/router/BridgeQueue.sol

```diff
@@ -0,0 +1,600 @@
+            )
+        returns (bool routerSupported) {
+            if (!routerSupported) revert InvalidBridgeRouter();
+        } catch {
+            revert InvalidBridgeRouter();
+        }
+
+        bridgeRouter = _newBridgeRouter;
+        emit BridgeRouterUpdated(_newBridgeRouter);
+    }
+
+    /// @inheritdoc IBridgeQueue
+    function addQueueManager(address manager) external onlyGovernor {
+        if (manager == address(0)) revert InvalidQueueManager();
+        if (!isQueueManager[manager]) {
+            isQueueManager[manager] = true;
+            emit QueueManagerAdded(manager);
+        }
+    }
+
+    /// @inheritdoc IBridgeQueue
+    function removeQueueManager(address manager) external onlyGovernor {
+        if (manager == address(0)) revert InvalidQueueManager(); // Also check for zero address on removal
+        if (isQueueManager[manager]) {
+            isQueueManager[manager] = false;
+            emit QueueManagerRemoved(manager);
+        }
+    }
+
+    /// @inheritdoc IBridgeQueue
+    function dequeueOperation(
+        bytes32 queueId
+    ) external onlyGovernor nonReentrant {
+        uint256 index = _pendingQueueIdIndex[queueId];
+        if (index == 0) revert QueueIdNotFound();
+        if (queueIdToStatus[queueId] != BridgeTypes.OperationStatus.QUEUED)
+            revert OperationNotQueued();
+
+        // If this is a transfer operation, notify the originator to reset inflight assets
+        if (
+            queueIdToOperationType[queueId] ==
+            BridgeTypes.OperationType.TRANSFER_ASSET
+        ) {
+            address originator = queuedTransfers[queueId].originator;
+            // Check if originator supports IInflightAssetTracking interface before calling updateInflightAssets
+            if (originator.code.length > 0) {
+                try
+                    IERC165(originator).supportsInterface(
+                        type(IInflightAssetTracking).interfaceId
+                    )
+                returns (bool supported) {
+                    if (supported) {
+                        try
+                            IInflightAssetTracking(originator)
+                                .updateInflightAssets(0)
+                        {} catch {
+                            // Ignore failures in updateInflightAssets
+                        }
+                    }
+                } catch {
+                    // Originator doesn't support ERC165 or IInflightAssetTracking, ignore
+                }
+            }
+        }
+
+        _removePendingId(queueId, index - 1);
+        queueIdToStatus[queueId] = BridgeTypes.OperationStatus.FAILED; // Mark as failed since dequeued
+
+        emit OperationDequeued(queueId, msg.sender); // Governor initiated dequeue
+    }
+
+    /// @inheritdoc IBridgeQueue
+    function recoverFunds(
+        address token,
+        address recipient,
+        uint256 amount
+    ) external onlyGovernor {
```

Comment:
question is how often would it happen - if it's predicted it might happen often, that will be bad UX - with funds stuck for weeks, plus the gov uncertainity